File: chor-extraction/src/main/java/ast/cc/nodes/Communication.java
package ast.cc.nodes;

import ast.cc.CCVisitor;
import ast.cc.interfaces.CCNode;
import ast.cc.interfaces.Interaction;
import ast.sp.nodes.expr.BooleanExpression;
import epp.MergingException;

/**
 * Created by lara on 03/04/17.
 */
public class Communication implements Interaction
{
    private final String sender;
    private final String receiver;
    private final BooleanExpression expression;
    private final CCNode continuation;

    public CCNode getContinuation() {
        return continuation;
    }

    public Communication(String sender, String receiver, BooleanExpression expression, CCNode continuation) {
        this.sender = sender;
        this.receiver = receiver;
        this.expression = expression;
        this.continuation = continuation;
    }

    public <T> T accept( CCVisitor<T> visitor ) throws MergingException {
        return visitor.visit( this );
    }

    public String getSender() {
        return sender;
    }

    public String getReceiver() {
        return receiver;
    }

    public BooleanExpression getExpression() {
        return expression;
    }

    public String toString(){
        return sender + "." + expression + "->" + receiver + "; " + continuation.toString();
    }
}

File: chor-extraction/src/main/java/ast/cc/nodes/Communication.kt
package ast.cc.nodes

import ast.cc.CCVisitor
import ast.cc.interfaces.CCNode
import ast.cc.interfaces.Interaction
import epp.MergingException

data class Communication(val sender: String, val receiver: String, val expression: String, val continuation: CCNode) : Interaction {

    @Throws(MergingException::class)
    override fun <T> accept(visitor: CCVisitor<T>): T {
        return visitor.visit(this)
    }

    override fun toString(): String {
        return sender + "." + expression + "->" + receiver + "; " + continuation.toString()
    }
}
